states/new_hampshire/main.py

```diff
@@ -0,0 +1,84 @@
+from common import work_dir
+import json
+import scrapy
+from scrapy.crawler import CrawlerProcess
+import csv
+import re
+from io import StringIO
+import numpy as np
+
+# We use Scrapy for New Hampshire because it has an .aspx __viewstate which must be generated and sent along with the request
+
+
+class NewHampshireSpider(scrapy.Spider):
+    name = 'New Hampshire'
+    start_urls = ["https://app.sos.nh.gov/Public/Reports.aspx"]
+
+    def parse(self, response):
+        yield scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                'ctl00$MainContentPlaceHolder$PPReport': 'rdoCsv'
+            },
+            method="POST",
+            callback=self.generate_csv
+        )
+
+    def generate_csv(self, response):
+        f = StringIO(response.text)
+        reader = csv.DictReader(f, delimiter=",")
+        clerk_data = []
+        for row in reader:
+            town_or_city = " ".join([x.lower().capitalize()
+                                     for x in row["Town/City"].split(" ")])
+            email_pattern = re.compile(
+                r"^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$")
+
+            if(not email_pattern.match(row["E-Mail"]) or not row["Fax"]):
+                data = {
+                    "city": town_or_city,
+                    "locale": town_or_city,
+                    "address": row["Address"],
+                    "emails" : [],
+                    "faxes" : [],
+                }
+            else:
+                data = {
+                    "city": town_or_city,
+                    "official":  row["Clerk"],
+                    "emails": [row["E-Mail"]],
+                    "phones": ["603-" + row["Phone (area code 603)"]],
+                    "locale": town_or_city,
+                    "faxes": ["603-" + row["Fax"]],
+                    "address": row["Address"],
+                }
+
+            clerk_data.append(data)
+
+        deduped_data = self.dedupe_by_ward(clerk_data)
+
+        with open('../../public/new_hampshire.json', 'w') as f:
+            json.dump(deduped_data, f)
+
+        return clerk_data
+
+    def dedupe_by_ward(self, clerk_data):
+        distinct_logical_cities = set(x["city"][0:x["city"].find(
+            " Ward ")] for x in clerk_data if " Ward " in x["city"])
+        deduped_clerks = [
```

Comment:
The elements of `clerk_data` should always be the same.  If we use `x` in the line above, we should use `x` in this line.